test/src/kbasesearchengine/test/main/SearchMethodsTest.java

```
@@ -669,4 +672,120 @@ private void compare(final SortingRule got, final SortingRule expected) {
+        privateData.put("test", 1L);
+
+        final Map<String, Long> allData = new HashMap<>();
+        allData.put("test", 1L);
+        allData.put("notdefault", 1L);
+
+        final Map<String, Long> publicData = new HashMap<>();
+        publicData.put("notdefault", 1L);
+
+
+        final SearchTypesOutput res = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(1L).withWithPublic(1L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+
+        assertThat("should show both public and private", res.getTypeToCount(), is(allData));
+
+
+        final SearchTypesOutput res2 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(0L).withWithPublic(1L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only public", res2.getTypeToCount(), is(publicData));
+
+        final SearchTypesOutput res3 = sm.searchTypes(new SearchTypesInput()
+                        .withAccessFilter(new AccessFilter().withWithPrivate(1L).withWithPublic(0L))
+                        .withMatchFilter(new MatchFilter()),
+                "auser");
+        assertThat("should show only private", res3.getTypeToCount(), is(privateData));
+
+        //should not happen
+        try{
+            final SearchTypesOutput res4 = sm.searchTypes(new SearchTypesInput()
+                            .withAccessFilter(new AccessFilter().withWithPrivate(0L).withWithPublic(0L))
+                            .withMatchFilter(new MatchFilter()),
+                    "auser");
+        }catch(IllegalArgumentException e){
+            assertThat("should throw exception if private and public set to false"
+                    , e.getMessage(), is("should have either public or private data"));
+        }catch(Exception e){
```

You don't want to do this, this will mask other exceptions that might occur.